pinc/Activity.inc

```diff
@@ -41,31 +60,40 @@ class Activities
         "Pools" => "Pool",
     ];
 
-    private static function _get_class_mapping()
+    private static function _get_class_mapping(): string
     {
         return self::$container_to_class_mapping[static::class];
     }
 
-    public static function add(&$activity)
+    public static function add(Activity &$activity): void
     {
         foreach (array_merge([get_class($activity)], class_parents($activity)) as $class) {
             assert(!isset(self::$_activities[$class][$activity->id]), "Duplicate activity ID; IDs must be unique");
             self::$_activities[$class][$activity->id] = $activity;
         }
     }
 
-    public static function get_all()
+    /**
+     * The return of this function is untyped for now (see the comment above `$_activities` for why).
```

Comment:
This comment is likely to be used as the function description in the phpdocs that get generated. I think if we add these as a `//` style comment they won't. (also below)

Reply:
Done.